### PR TITLE
Workaround for the iss's pmp fault

### PR DESCRIPTION
**Pull request by silabs-krdosvik**
Signed-off-by: Kristine Dosvik <krdosvik@silabs.com>

There is a mismatch between the core and the ISS when NUM_PMP_REGIONS is zero. The ISS reports illegal instruction, while the core access fault. Because it is is a pmp fault the correct error code is access fault. This PR makes a workaround for this scenario.

Passes the ci check.

### CODE REVIEW

File: cv32e40s/tests/cfg/pmp_g3r3.yaml [hunk 1]
```diff
@@ -4,8 +4,170 @@ compile_flags:
...
 ovpsim: >
-    --override root/cpu/PMP_grain=3
-    --override root/cpu/PMP_registers=3
+    --override cpu/PMP_grain=3
+    --override cpu/PMP_registers=64
+    --override cpu/PMP_initialparams=T
+    --override cpu/PMP_maskparams=T
+    --override cpu/pmpaddr0=0
+    --override cpu/pmpaddr1=0
+    --override cpu/pmpaddr2=0
```
Comment (silabs-robin) [inline cv32e40s/tests/cfg/pmp_g3r3.yaml:13]: If there should be 3 regions in this config, should the first 3 overrides here not exist?

Comment (silabs-robin) [reply cv32e40s/tests/cfg/pmp_g3r3.yaml:13]: Or is that handled by the masks below?

File: cv32e40s/tests/cfg/clic_default.yaml [hunk 2]
```diff
@@ -4,17 +4,170 @@ compile_flags:
...
-    --override cpu/CLICXCSW=T
-    --override cpu/CLICXNXTI=T
-    --override cpu/CLIC_version=master
-    --override cpu/externalCLIC=T
-    --override cpu/Zcmb=f
-    --override cpu/Zicsr=T
-    --override cpu/noinhibit_mask=0xfffffffa
-    --override cpu/trigger_num=0
-    --override cpu/extension/mcountinhibit_reset=0x5
-    --override cpu/mtvec_mask=0xffffff80
```
Comment (silabs-hfegran) [inline cv32e40s/tests/cfg/clic_default.yaml:17]: These should not have been removed

File: cv32e40s/tests/cfg/pmp_g3r3.yaml (hunk 1 above)
Comment (silabs-hfegran) [reply cv32e40s/tests/cfg/pmp_g3r3.yaml:13]: pmpaddrX sets default value, mask_pmpaddrX sets writeable bitmask. The RTL expects all 64 registers to always be present, even when PMP number of regions is set to 0 - this fix works around that in the ISS, which completely disables the PMP regions if number of regions = 0

**Review by silabs-hfegran — CHANGES_REQUESTED** [review]

File: cv32e40s/tests/cfg/pmp_g3r3.yaml (hunk 1 above)
Comment (silabs-robin) [reply cv32e40s/tests/cfg/pmp_g3r3.yaml:13]: Good

File: cv32e40s/tests/cfg/clic_default.yaml (hunk 2 above)
Comment (silabs-krdosvik) [reply cv32e40s/tests/cfg/clic_default.yaml:17]: True, reinserted!

**Review by silabs-hfegran — APPROVED** [review]
I am approving this for now - should be moved into the common configuration object when things stabilize

**Pull request merged**